tests/test_path.py

```diff
@@ -117,3 +123,179 @@ def test_find_case_sensitive_path(tmp_path, path, existing_paths, expected):
 
     result = find_case_sensitive_path(tmp_path / path, sys.platform)
     assert result == tmp_path / expected
+
+
+@pytest.fixture()
+def simple_module(tmp_path: Path) -> Path:
+    fn = tmp_path / "_src/tests/mymod.py"
+    fn.parent.mkdir(parents=True)
+    fn.write_text("def foo(x): return 40 + x")
+    return fn
+
+
+def test_importmode_importlib(simple_module: Path, tmp_path: Path) -> None:
+    """`importlib` mode does not change sys.path."""
+    module = import_path(simple_module, root=tmp_path)
+    assert module.foo(2) == 42  # type: ignore[attr-defined]
+    assert str(simple_module.parent) not in sys.path
+    assert module.__name__ in sys.modules
+    assert module.__name__ == "_src.tests.mymod"
+    assert "_src" in sys.modules
+    assert "_src.tests" in sys.modules
+
+
+def test_importmode_twice_is_different_module(
+    simple_module: Path, tmp_path: Path
+) -> None:
+    """`importlib` mode always returns a new module."""
+    module1 = import_path(simple_module, root=tmp_path)
+    module2 = import_path(simple_module, root=tmp_path)
+    assert module1 is not module2
+
+
+def test_no_meta_path_found(
+    simple_module: Path, monkeypatch: pytest.MonkeyPatch, tmp_path: Path
+) -> None:
+    """Even without any meta_path should still import module."""
+    monkeypatch.setattr(sys, "meta_path", [])
+    module = import_path(simple_module, root=tmp_path)
+    assert module.foo(2) == 42  # type: ignore[attr-defined]
+
+    # mode='importlib' fails if no spec is found to load the module
+    import importlib.util
+
+    monkeypatch.setattr(
+        importlib.util, "spec_from_file_location", lambda *args: None  # noqa: ARG005
+    )
+    with pytest.raises(ImportError):
+        import_path(simple_module, root=tmp_path)
+
+
+def test_importmode_importlib_with_dataclass(tmp_path: Path) -> None:
+    """Ensure that importlib mode works with a module containing dataclasses (#7856)."""
```

Comment:
swap issue to https://github.com/pytask-dev/pytask/pull/373

Reply:
Or point to the absolute pytest issue https://github.com/pytest-dev/pytest/issues/7856